dbt/adapters/snowflake/relation_configs/target_lag.py

```diff
@@ -26,3 +39,101 @@ class SnowflakeDynamicTableTargetLagConfig(RelationConfigBase):
 
     duration: int
     period: SnowflakeDynamicTableTargetLagPeriod
+
+    @property
+    def validation_rules(self) -> Set[RelationConfigValidationRule]:
+        return {
+            RelationConfigValidationRule(
+                validation_check=self.duration is not None and self.period is not None,
+                validation_error=DbtRuntimeError(
+                    "Target lag for a materialized view requires both a duration and a period."
+                ),
+            ),
+            RelationConfigValidationRule(
+                validation_check=(
+                    (
+                        self.duration >= 1
+                        and self.period != SnowflakeDynamicTableTargetLagPeriod.seconds
+                    )
+                    or (
+                        self.duration >= 60
+                        and self.period == SnowflakeDynamicTableTargetLagPeriod.seconds
+                    )
+                ),
+                validation_error=DbtRuntimeError(
+                    f"The minimum for target lag for a materialized view is 1 minute. "
+                    f"The provided value is {self.duration} {self.period.value}."
+                ),
+            ),
+        }
+
+    @classmethod
+    def from_dict(cls, config_dict: dict) -> "SnowflakeDynamicTableTargetLagConfig":
+        kwargs_dict: Dict[str, Union[int, SnowflakeDynamicTableTargetLagPeriod]] = {}
+
+        if duration := config_dict.get("duration"):
+            kwargs_dict.update({"duration": int(duration)})
+
+        if period := config_dict.get("period"):
+            kwargs_dict.update({"period": SnowflakeDynamicTableTargetLagPeriod(period)})
+
+        target_lag: "SnowflakeDynamicTableTargetLagConfig" = super().from_dict(kwargs_dict)  # type: ignore
+        return target_lag
+
+    @classmethod
+    def parse_model_node(cls, model_node: ModelNode) -> dict:
```

Comment:
I don't understand this paradigm you've made but I want to! Who is calling this? Why is the name so generic when the result is so specific to the class? (I see you do this elsewhere)

Reply:
It's a poor man's `mashumaro`, `pydantic`, `marshmallow`, etc. I could see replacing this in the long term with a framework like that. The idea is that there are several ways to create this "generic" dynamic view representation. This dataclass is just one giant set of parsers with validation rules. The two parsers I found so far is database > config and user-config > config. I wouldn't be surprised if we wind up with more. Maybe someone wants to get crazy and do a Redshift MV user-config > Snowflake DT config to do a migration. Whatever it is, this config class is our anchor. And we generally want two representations of this config. We want a nice shiny, python object with dot path-ing, etc. so that the jinja reads cleanly, other dependent code reads cleanly, etc. And then we usually want the dictionary version that is ubiquitous; it's nice for logs, it translates out of python nicely if necessary (basically json). In order to do that in an hierarchical object, each object needs both and we need to split apart the foreign format > dict and dict > object computation. A framework like `mashumaro` does all of that behind the scenes so you don't need to worry about it. However, we still have the added complexity of needing more than one parser since our objects come from more than one source. I'd be happy to discuss this further, I think it need sync discussions to make sense.